pkgs/development/python-modules/libmambapy/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, fetchFromGitHub
+, buildPythonPackage
+, cmake
+, ninja
+, libmamba
+, pybind11
+, setuptools
+, wheel
+}:
+buildPythonPackage rec {
+  pname = "libmambapy";
+  version = "1.5.7";
+  pyproject = true;
+
+  nativeBuildInputs = [ cmake ninja ];
+  buildInputs = [ libmamba pybind11 ];
+
+  src = fetchFromGitHub {
+    owner = "mamba-org";
+    repo = "mamba";
+    rev = "${pname}-${version}";
+    hash = "sha256-HfmvLi9IBWlaGAn2Ej4Bnm4b3l19jEXwNl5IUkdVxi0=";
+  };
+
+  souceRoot = "${src.name}/libmambapy";
+  cmakeFlags = [ "-GNinja" (lib.cmakeBool "BUILD_LIBMAMBAPY" true) ];
+
+  buildPhase = ''
+    runHook preBuild
+    ninjaBuildPhase
+    cp -r libmambapy ../libmambapy
+    cd ../libmambapy
+    pypaBuildPhase
+    runHook postBuild
```

Comment:
Which hook specifically?
I understood and see examples that to override a phase, the `prePhase` and the `postPhase` should be called.

Reply:
Oh, ignore me, I'm confused. That said, you can remove `runHook preBuild` and `runHook postBuild` because it's already called twice by `ninjaBuildPhase` and by `pypaBuildPhase`...

Reply:
(which means that this phase is only "correct" if the hooks are idempotent)